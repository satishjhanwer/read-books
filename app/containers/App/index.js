/**
 *
 * App
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 */

import React from 'react';
import { Helmet } from 'react-helmet';
import { Route, Routes } from 'react-router-dom';
import { Container, Row, Col } from 'reactstrap';

import Navigation from '../../components/Navigation/Loadable';
import HomePage from '../HomePage/Loadable';
import AuthorDetailPage from '../AuthorDetails/Loadable';
import BookDetailPage from '../BookDetails/Loadable';
import NotFoundPage from '../NotFoundPage/Loadable';

export default function App() {
  return (
    <>
      <Helmet titleTemplate="%s - Search Books" defaultTitle="Search Books">
        <meta name="description" content="A React.js application for searching books" />
      </Helmet>
      <Navigation />
      <Container>
        <Row>
          <Col sm="12">
            <Routes>
              <Route exact path="/" element={<HomePage />} />
              <Route path="/book/:id" element={<BookDetailPage />} />
              <Route path="/author/:id" element={<AuthorDetailPage />} />
              <Route path="*" element={<NotFoundPage />} />
            </Routes>
          </Col>
        </Row>
      </Container>
    </>
  );
}
