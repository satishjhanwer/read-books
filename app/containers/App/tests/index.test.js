import React from 'react';
import { render } from '@testing-library/react';

import App from '../index';

describe('<App />', () => {
  it('should render and match the snapshot', () => {
    const renderedOutput = render(<App />);
    expect(renderedOutput).toMatchSnapshot();
  });
});
