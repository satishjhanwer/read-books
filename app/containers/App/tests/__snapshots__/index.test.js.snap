// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<React.Fragment>
  <HelmetWrapper
    defaultTitle="Search Books"
    defer={true}
    encodeSpecialCharacters={true}
    titleTemplate="%s - Search Books"
  >
    <meta
      content="A React.js application for searching books"
      name="description"
    />
  </HelmetWrapper>
  <Unknown />
  <Container
    tag="div"
  >
    <Row
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Col
        sm="12"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/book/:id"
          />
          <Route
            component={[Function]}
            path="/author/:id"
          />
          <Route
            component={[Function]}
            path=""
          />
        </Switch>
      </Col>
    </Row>
    <GlobalStyleComponent />
  </Container>
</React.Fragment>
`;
